spec/enumerable_spec.rb

```diff
@@ -0,0 +1,296 @@
+        expect(array.my_each_with_index { |_item, index| index + 1 }).to eql(expected)
+      end
+      it 'loops to each index' do
+        expected = array.each_with_index { |_item, index| index * 1 }
+        expect(array.my_each_with_index { |_item, index| index * 1 }).to eql(expected)
+      end
+    end
+  end
+
+  describe '#my_select' do
+    context 'when no block is given' do
+      it 'returns enum when no block is given' do
+        expect(array.my_select).to be_an Enumerator
+      end
+    end
+    context 'when block is given' do
+      it 'returns array containing the matching elements' do
+        expected = array.select(&:even?)
+        expect(array.my_select(&:even?)).to eql(expected)
+      end
+      it 'returns array containing the matching elements of an operation' do
+        expected = array.select { |item| item % 3 == 0 }
+        expect(array.my_select { |item| item % 3 == 0 }).to eql(expected)
+      end
+    end
+  end
+
+  describe '#my_all?' do
+    context 'when no block is given' do
+      it 'items are not false or nil' do
+        expect(array_string.my_all?).to eql(array_string.all?)
+      end
+      it 'items are false or nil' do
+        expect(array_with_nil.my_all?).to eql(array_with_nil.all?)
+      end
+      it 'items are not given' do
+        expect([].my_all?).to eql([].all?)
+      end
+    end
+    context 'when block is given' do
+      it 'items are not false or nil' do
+        expected = array_string.all? { |word| word.length >= 3 }
+        expect(array_string.my_all? { |word| word.length >= 3 }).to eql(expected)
+      end
+      it 'items are not false or nil' do
+        expected = array_string.all? { |word| word.length >= 4 }
+        expect(array_string.my_all? { |word| word.length >= 4 }).to eql(expected)
+      end
+    end
+    context 'when regular expression is given' do
+      it 'items are not false or nil' do
+        expected = array_string.all?(/t/)
+        expect(array_string.my_all?(/t/)).to eql(expected)
+      end
+    end
+    context 'when class is given' do
+      it 'items are not false or nil' do
+        expected = array_with_invalid_numeric.all?(Numeric)
+        expect(array_with_invalid_numeric.my_all?(Numeric)).to eql(expected)
+      end
+    end
+  end
+
+  describe '#my_any?' do
+    context 'when no block is given' do
+      it 'returns false' do
+        expect([].my_any?).to eql(false)
+      end
+      it 'items are false or nil' do
+        expect(array_with_nil.my_any?).to eql(array_with_nil.any?)
+      end
+    end
+    context 'when block is given' do
+      it 'items are not false or nil' do
+        expected = array_string.any? { |word| word.length >= 3 }
+        expect(array_string.my_any? { |word| word.length >= 3 }).to eql(expected)
+      end
+      it 'items are not false or nil' do
+        expected = array_string.any? { |word| word.length >= 4 }
+        expect(array_string.my_any? { |word| word.length >= 4 }).to eql(expected)
```

Comment:
Messages should be more clear. So check the provided messages and you can modify it to update for this one.

Reply:
Fixed, thanks for the suggestion